dmutils/filters.py

```diff
@@ -24,19 +25,23 @@ def format_links(text):
                                 \.                        # match dot in host name
                                 (?:[^\s/?\.#<>"']+\.?)+   # match rest of domain
                                 (?:/[^\s<>"']*)?          # rest of url
-                                [^\s<>"'\.]               # no dot at end
+                                [^\s<>,"'\.]              # no dot at end
                                 )""", re.X)
-    matches = url_match.findall(text)
-    link = '{0}<a href={1} rel="external">{1}</a>'
-    if matches:
-        textArray = url_match.split(text)
-        formattedText = ""
-        for text in textArray:
-            if text in matches:
-                formattedText = link.format(formattedText, Markup.escape(text))
+    matched_urls = url_match.findall(text)
+    if matched_urls:
+        link = '<a href="{0}" rel="external">{1}</a>'
+        formatted_text = ""
+        text_array = url_match.split(text)
+        formatted_text_array = []
+        for partial_text in text_array:
+            if partial_text in matched_urls:
+                url = "https://{}".format(partial_text) if partial_text.startswith('www') else partial_text
```

Comment:
I think it might actually be okay.  When you visit it without it gives you warning and also generally sites should be https these days. There was another way of doing it by putting // in front but this is advised against now and the suggestion was to always make it https instead.

I was going to ask about the http being promoted to https problem though when Alexey got back as I can't really find anything on it easily.

The problem with dropping www is that this is being put in so we could still end up with the same problem of the page not being readable because some weird long url has been put in.  I think that if we really can't make it into a link that it might actually be better to put a span around it so that it can still have the class that allows it break.